code/game/objects/items/dehy_carp.dm

```diff
@@ -36,16 +37,16 @@
 	//Wait for animation to end
 	sleep(6)
 	//Make space carp
-	var/mob/living/simple_animal/hostile/carp/C = new /mob/living/simple_animal/hostile/carp(get_turf(src))
+	var/mob/living/M = new mobtype(get_turf(src))
 	//Make carp non-hostile to user, and their allies
 	if(owner)
 		var/list/factions = owner.faction
 		for(var/F in factions)
 			if(F == "neutral")
 				factions -= F
-		C.faction = factions
-	if (!owner || owner.faction != C.faction)
+		M.faction = factions
+	if (!owner || owner.faction != M.faction)
 		visible_message("<span class='warning'>You have a bad feeling about this.</span>") //welcome to the hostile carp enjoy your die
 	else
-		visible_message("<span class='notice'>The newly grown carp looks up at you with friendly eyes.</span>")
+		visible_message("<span class='notice'>The newly grown [M.name] looks up at you with friendly eyes.</span>")
```

Comment:
You branched off of your old carp PR instead of off of master. Use "git checkout upstream/master -- [filename]" to reset accidentally modified files like this.